### PR TITLE
feat(pin): eval constraints while pinned, but don't approve anything

### PR DESCRIPTION
**Pull request by emjburns**
fixes https://github.com/spinnaker/keel/issues/1254

omg this was so easy to do now that i re-wrote everything.

Note: this pr removes the old environment promotion checker tests as well, in favor of the new ones. I didn't rename the new ones because it makes the pr super hard to understand.

### CODE REVIEW

File: keel-core/src/main/kotlin/com/netflix/spinnaker/keel/actuation/EnvironmentPromotionChecker.kt [hunk 1]
```diff
@@ -40,24 +40,22 @@ class EnvironmentPromotionChecker(
...
-              constraintRunner.checkEnvironment(
-                EnvironmentContext(
-                  deliveryConfig = deliveryConfig,
-                  environment = environment,
-                  artifact = artifact,
-                  versions = versions,
-                  vetoedVersions = (vetoedArtifacts[envPinKey(environment.name, artifact)]?.versions) ?: emptySet()
-                )
-              )
+              constraintRunner.checkEnvironment(envContext)
```
Comment (gal-yardeni) [inline keel-core/src/main/kotlin/com/netflix/spinnaker/keel/actuation/EnvironmentPromotionChecker.kt:58]: I think you can remove the `else` here, as constraintRunner.checkEnvironment(envContext) is being called in both cases?

Comment (emjburns) [reply keel-core/src/main/kotlin/com/netflix/spinnaker/keel/actuation/EnvironmentPromotionChecker.kt:58]: The important thing is in the pin case, it's _only_ checking the environments and not doing anything else. So the `else` needs to stay

File: keel-core/src/test/kotlin/com/netflix/spinnaker/keel/actuation/NewEnvironmentPromotionCheckerTests.kt [hunk 2]
```diff
@@ -201,19 +201,24 @@ internal class NewEnvironmentPromotionCheckerTests : JUnit5Minutests {
...
-          test("the artifact is approved") {
+          test("the pinned artifact is approved") {
             verify(exactly = 1) {
               repository.approveVersionFor(deliveryConfig, artifact, "1.0", environment.name)
             }
           }
+
+          test("queued constraints aren't looked at") {
+            verify(exactly = 0) {
+              environmentConstraintRunner.checkStatelessConstraints(any(), any(), any(), any())
```
Comment (lorin) [inline keel-core/src/test/kotlin/com/netflix/spinnaker/keel/actuation/NewEnvironmentPromotionCheckerTests.kt:218]: This is a very small nit, but it looks to me like this test is testing two distinct things, stateless constraints and queued constraints. I would pull out the stateless constraint check into a separate test for clarity, e.g.:

``` kotlin
          test("stateless constraints aren't looked at") {
            verify(exactly = 0) {
              environmentConstraintRunner.checkStatelessConstraints(any(), any(), any(), any())
            }
          }

```

File: keel-core/src/main/kotlin/com/netflix/spinnaker/keel/actuation/EnvironmentPromotionChecker.kt (hunk 1 above)
Comment (gal-yardeni) [reply keel-core/src/main/kotlin/com/netflix/spinnaker/keel/actuation/EnvironmentPromotionChecker.kt:58]: I guess reading code via GH is hard lol. thanks for clarifying :)

**Pull request merged**